registry/src/pages/account.js

```diff
@@ -1,13 +1,253 @@
-import React from "react";
-import { Navigate } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { useDispatch, useSelector } from "react-redux";
 import { useCookies } from "react-cookie";
+import { fetchPackages } from "../store/actions/dashboardActions";
+import { useNavigate } from "react-router-dom";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import { MDBIcon } from "mdbreact";
+import { MDBBtn } from 'mdb-react-ui-kit';
+import Image from "react-bootstrap/Image";
+import Table from "react-bootstrap/Table";
+import Spinner from "react-bootstrap/Spinner";
+import Container from "react-bootstrap/Container";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+import "@fortawesome/fontawesome-free/css/all.min.css";
+import "bootstrap-css-only/css/bootstrap.min.css";
+import "mdbreact/dist/css/mdb.css";
 
 const Account = () => {
+  const [email, setEmail] = useState("");
+  const [error, setError] = useState("");
   const [cookies, setCookie] = useCookies(["uuid"]);
-  if (!cookies.uuid) {
-    return <Navigate to="/account/login" replace={true} />;
-  }
-  return <p>Account</p>;
+  const [password, setPassword] = useState("");
+  const [oldpassword, setOldpassword] = useState("");
+  const [dateJoined, setdateJoined] = useState("");
+  const username = useSelector((state) => state.auth.username);
+  const uuid = useSelector((state) => state.auth.uuid);
+  const packages = useSelector((state) => state.dashboard.packages);
+  const isLoading = useSelector((state) => state.dashboard.isLoading);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (username === null) {
+      navigate("/");
+    } else if (packages === null) {
+      dispatch(fetchPackages(username));
+
+      let formData = new FormData();
+      formData.append("uuid", uuid);
+
+      axios({
+        method: "post",
+        url: `${process.env.REACT_APP_REGISTRY_API_URL}/users/account`,
+        data: formData,
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      })
+        .then((result) => {
+          setEmail(result.data.user.email);
+          setdateJoined(result.data.user.createdAt.slice(0, 16));
+        })
+        .catch((error) => {
+          console.log(cookies.uuid);
+        });
+    }
+  });
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    let formData = new FormData();
+    formData.append("oldpassword", oldpassword);
+    formData.append("password", password);
+    formData.append("uuid", uuid);
+
+    axios({
```

Comment:
This also should be inside `accountActions.js` file. You should create an `accountReducer.js` file as well to emit the states accordingly and use `useSelector` hook to get the states.

Reply:
sure.